### PR TITLE
Change metacard interactions to use permissions hook

### PR DESCRIPTION
**Pull request by samuelechu**
Also adds context to workspaces so that security attributes can be passed down for query cards to show available actions. https://github.com/connexta/revelio/pull/333/commits/904b35a44d5821b60a3ac11e81e542bda775d188

### CODE REVIEW

File: src/main/webapp/components/result-forms/route.js [hunk 1]
```diff
@@ -91,7 +73,7 @@ const Item = props => {
         </Dialog>
       ) : null}
       <IndexCardItem {...form} onClick={() => setEditing(true)}>
-        <Actions attributes={form} user={user}>
+        <Actions attributes={form}>
```
Comment (samuelechu) [inline src/main/webapp/components/result-forms/route.js:76]: Going to see if I can use context in a future pr which will allow a user to be passed through to the hook in storybook

File: src/main/webapp/components/workspace/workspace.js [hunk 2]
```diff
@@ -138,121 +144,123 @@ export default () => {
...
   const { title } = attributes
   return (
-    <div
-      style={{
-        display: 'flex',
-        height: 'calc(100vh - 64px)',
-        overflow: 'hidden',
-      }}
-    >
+    <WorkspaceContext.Provider value={attributes}>
```
Comment (samuelechu) [inline src/main/webapp/components/workspace/workspace.js:147]: The only thing that changed below this line is the closing tag of WorkspaceContext.Provider

File: src/main/webapp/components/search-forms/search-forms.stories.tsx [hunk 3]
```diff
@@ -44,10 +44,10 @@ const startingForms = [
...
 ]
 
-const userAttributes = {
-  email: 'admin@test.com',
-  roles: ['admin'],
-}
+// const userAttributes = {
+//   email: 'admin@test.com',
+//   roles: ['admin'],
+// }
```
Comment (willwill96) [inline src/main/webapp/components/search-forms/search-forms.stories.tsx:50]: delete?

File: src/main/webapp/react-hooks/use-get-permissions.tsx [hunk 4]
```diff
@@ -19,12 +26,12 @@ const user = gql`
...
-const usePermissions = (user: User) => {
-  return (attributes: any) => {
+const usePermissions = (user: User = {}) => {
+  return (attributes: any = {}) => {
     const securityAttributes = getSecurityAttributesFromMetacard(attributes)
     return getPermissions(
-      user.email,
-      user.roles,
+      user.email || '',
+      user.roles || [],
```
Comment (willwill96) [inline src/main/webapp/react-hooks/use-get-permissions.tsx:34]: with these changes do you still need, this change? https://github.com/connexta/revelio/pull/333/files#diff-8e61c7dfd17cef5f94afbba276fdbc9eR56

Comment (samuelechu) [reply src/main/webapp/react-hooks/use-get-permissions.tsx:34]: An undefined value still may be returned if the user query is still loading https://github.com/connexta/revelio/pull/333/files/e8340ec91a66c723502d36849e1ebd797cb8eabd#diff-fc3b9df5837400effc9fe86ea4d16242R44

File: src/main/webapp/components/search-forms/search-forms.stories.tsx (hunk 3 above)
Comment (samuelechu) [reply src/main/webapp/components/search-forms/search-forms.stories.tsx:50]: I'm hoping to uncomment this later, once I have the usePermissions hook able to take in a user through context

**Comment by samuelechu** [comment]
> I think user still needs to be removed from the workspaces.gql file as well as the workspaces use-create hook

This is still using user. Do you think it would be good to create a hook for just user?https://github.com/connexta/revelio/blob/master/src/main/webapp/components/workspaces/workspaces.js#L164

**Pull request merged**